### PR TITLE
Allow to sign past/future certificates

### LINKED ISSUE #96: Can not sign a certificate out of notBefore and notAfter validity.
The "sanity check" in `CertificateAuthority::_signCSR()` does not allow me to sign a certificate with expired validity or a certificate that will be valid in the future, because validation at the moment must succeed at the end of the function. Anything that is not valid at the moment of the signature can not be signed.

### PR DESCRIPTION
**Pull request by miroshko**
within validity bounds of the root certificate

Resolves #96

### CODE REVIEW

File: src/x509.cpp [hunk 1]
```diff
@@ -211,7 +211,10 @@ void X509Certificate::verify(const std::vector<X509Certificate> &trustStore,
 {
     VerificationContext ctx;
     ctx.addTrustedCertificates(trustStore)
-       .addIntermediateCertificates(intermediateCAs);
+       .addIntermediateCertificates(intermediateCAs)
+       .setVerificationCheckTime(getNotBeforeAsn1());
+    verify(ctx);
+    ctx.setVerificationCheckTime(getNotAfterAsn1() - Asn1Time::Seconds(1));
```
Comment (miroshko) [inline src/x509.cpp]: still need to clarify this -1 second. according to the name and to the brief glance at the openssl code this shouldn't be needed

File: src/ca.cpp [hunk 2]
```diff
@@ -98,7 +98,13 @@ X509Certificate CertificateAuthority::_signCSR(const CertificateSigningRequest &
     //Sanity check: certificate must be verifiable now
     try {
         auto cert = X509Certificate{std::move(newCertificate)};
-        cert.verify({_rootCert}, {});
+        X509Certificate::VerificationContext ctx;
+        ctx.addTrustedCertificates({_rootCert})
+            .addIntermediateCertificates({})
+            .setVerificationCheckTime(cert.getNotBeforeAsn1());
+        cert.verify(ctx);
+        ctx.setVerificationCheckTime(cert.getNotAfterAsn1() - Asn1Time::Seconds(1));
```
Comment (garw) [inline src/ca.cpp:108]: If we need this, please add a comment due to the asymmetrie with NotBefore. However, I'd also think that it is not necessary.

**Review by garw — CHANGES_REQUESTED** [review]
Clearify the notAfter issue. 
Otherwise this looks good to me. Tests look also good. Wanted to see the CA / cert overlap problem but it is included :)

Please update the release notes to include this (so far unreleased change).

File: src/ca.cpp (hunk 2 above)
Comment (miroshko) [reply src/ca.cpp:108]: I added a comment, that's indeed how OpenSSL checks it

**Review by garw — APPROVED** [review]

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,14 @@ All notable changes to this project will be documented in this file.
 
 ## Fixed
 
+## Added
+
+# Release 4.1.1
+
+## Changed
+
```
Comment (neverpanic) [inline CHANGELOG.md:16]: I'd drop headlines with no content from the release notes.

**Review by neverpanic — COMMENTED** [review]
Nitpick, lg2m otherwise.

**Review by wusto — APPROVED** [review]

**Pull request merged**